payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncher.kt

```diff
@@ -1,95 +1,188 @@
+            GooglePayLauncherContract()
+        ) {
+            callback.onResult(it)
+        }
     )
 
     /**
-     * Constructor to be used when launching [GooglePayLauncher] from an Activity.
+     * Constructor to be used when launching [GooglePayLauncher] from a Fragment.
      *
      * @param fragment the Fragment that is launching the [GooglePayLauncher]
+     *
+     * @param readyCallback called after determining whether Google Pay is available and ready on
+     * the device. [present] may only be called if Google Pay is ready.
+     *
      * @param callback called with the result of the [GooglePayLauncher] operation
      */
     internal constructor(
         fragment: Fragment,
+        config: Config,
+        readyCallback: ReadyCallback,
         callback: ResultCallback
     ) : this(
-        { fragment.viewLifecycleOwner.lifecycleScope },
-        DefaultGooglePayController(
-            fragment,
-            googlePayRepositoryFactory = {
-                DefaultGooglePayRepository(
-                    fragment.requireActivity().application,
-                    it
-                )
-            },
-            callback = callback
-        )
+        fragment.viewLifecycleOwner.lifecycleScope,
+        config,
+        googlePayRepositoryFactory = {
+            DefaultGooglePayRepository(
+                fragment.requireActivity().application,
+                it
+            )
+        },
+        readyCallback,
+        fragment.registerForActivityResult(
+            GooglePayLauncherContract()
+        ) {
+            callback.onResult(it)
+        }
     )
 
-    fun configure(
-        configuration: GooglePayConfig,
-        callback: ConfigCallback
-    ) {
-        lifecycleScope().launch {
-            runCatching {
-                googlePayController.configure(configuration)
-            }.fold(
-                onSuccess = {
-                    callback.onConfigured(it, null)
-                },
-                onFailure = {
-                    callback.onConfigured(false, it)
+    init {
+        lifecycleScope.launch {
+            val repository = googlePayRepositoryFactory(config.environment)
+            readyCallback.onReady(
+                repository.isReady().first().also {
+                    isReady = it
                 }
             )
         }
     }
 
-    fun present() {
-        googlePayController.present()
-    }
+    fun present(clientSecret: String) {
+        check(isReady) {
+            "present() may only be called when Google Pay is available on this device."
+        }
 
-    fun interface ConfigCallback {
-        fun onConfigured(
-            success: Boolean,
-            error: Throwable?
+        activityResultLauncher.launch(
+            GooglePayLauncherContract.Args(
+                clientSecret = clientSecret,
+                config = config
+            )
         )
     }
 
+    @Parcelize
+    internal data class Config @JvmOverloads constructor(
```

Comment:
Can this still be `GooglePayConfig`?

Reply:
I was thinking of calling it that but Google Pay is implied because it will always be referenced as `GooglePayLauncher.Config`. This is more concise.